StandardLibraryEssentials/Templates/TypeAliases/task.md

```diff
@@ -0,0 +1,58 @@
+Type aliases are a feature in C++ that allows you to create a new name for an existing type. This can make your code more readable and easier to understand, especially when dealing with complex types such as templates or classes and functions deeply nested within namespaces. Type aliases can be created using either the `typedef` or `using` keywords.
+
+The `typedef` keyword, which has been in C++ since its first days, is used to create a new name for an existing type. The expression looks like this: ```typedef <current_type_name> <desired_type_name>;```. Here's how you can use `typedef` to create a type alias for `long long`:
+```cpp
+typedef long long ll;
+ll x = 1'234'567'890'987'654'321;
+```
+
+The `using` keyword, introduced in C++11, offers a more modern way to create type aliases. It is used in the same way, but the syntax is simpler and more readable than `typedef`. The format is: ```using <desired_type_name> = <current_type_name>;```. Here is how you can utilize `using` to create a type alias for `long long`:
+```cpp
+using ll = long long;
+ll x = 1'234'567'890'987'654'321;
+```
+
+On top of that, `using` can be employed to create type aliases for templates. This is especially useful when you want to create a type alias for a template with multiple parameters. Here's an example of how you can leverage `using` to create a type alias for a `std::vector` of `int` type (an upgraded version of an array, which will be covered in the next module):
+```cpp
+template <typename T>
+using Arr = std::vector<T>;
```

Comment:
Let's avoid using `std::vector` here, because it was not yet introduced. 
We can use `std::unique_ptr` as an example, because it was already mentioned before:

```
template <typename T>
using unique_array = std::unique_ptr<T[]>
```